sklearn/utils/_indexing.py

```diff
@@ -520,10 +527,18 @@ def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=
         )
 
     check_consistent_length(*arrays)
-
     if stratify is None:
         if replace:
-            indices = random_state.randint(0, n_samples, size=(max_n_samples,))
+            if sample_weight is not None:
+
+                sample_weight = sample_weight / sample_weight.sum()
+                indices = random_state.choice(
+                    np.arange(n_samples), size=(max_n_samples,), p=sample_weight
```

Comment:
I would rather use a different variable name for normalized weights. `sample_weight` is unnormalized in general, so better not have a variable name that can store either normalized or unnormalized weights at different place in the code.

```suggestion
                p = sample_weight / sample_weight.sum()
                indices = random_state.choice(
                    np.arange(n_samples), size=(max_n_samples,), p=p
```